crates/scroll/alloy/provider/src/engine/provider.rs

```diff
@@ -0,0 +1,135 @@
+use alloy_primitives::bytes::Bytes;
+use alloy_provider::{Network, RootProvider};
+use alloy_rpc_client::RpcClient;
+use alloy_rpc_types_engine::JwtSecret;
+use alloy_transport::utils::guess_local_url;
+use alloy_transport_http::{
+    hyper_util,
+    hyper_util::{client::legacy::connect::HttpConnector, rt::TokioExecutor},
+    AuthLayer, AuthService, Http, HyperClient,
+};
+use derive_more::Deref;
+use http_body_util::Full;
+use reqwest::Url;
+use scroll_alloy_network::Scroll;
+
+/// A JWT authenticated Http hyper client.
+type JwtAuthHttpClient<B = Full<Bytes>> =
+    Http<HyperClient<B, AuthService<hyper_util::client::legacy::Client<HttpConnector, B>>>>;
+
+/// An authenticated [`alloy_provider::Provider`] to the [`super::ScrollEngineApi`].
+#[derive(Debug, Clone, Deref)]
+pub struct ScrollAuthEngineApiProvider<N: Network = Scroll> {
```

Comment:
Given the amount of involved parts to get the authentification layer fitted in the client and finally all together in the `RootProvider`, I felt like having this encapsulated in a struct would facilitate usage.